### PR TITLE
[WIP][SPARK-34851][CORE][SQL] Add tag for each configuration

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add tag for each configuration

### Why are the changes needed?
Add tag for each configuration

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala [hunk 1]
```diff
@@ -279,3 +286,8 @@ private[spark] case class ConfigBuilder(key: String) {
     }
   }
 }
+
+object EffectTiming extends Enumeration {
+  type EffectTiming = Value
+  val LAUNCHING_DRIVER, DEPLOYING_APPLICATION, RUNTIME, DEPLOYING_CLUSTER, TESTING, UNKNOWN = Value
```
Comment (dohongdayi) [inline core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala:292]: Would you like to add comments to describe each of them, for me, I'm a little confuse on LAUNCHING_DRIVER, DEPLOYING_APPLICATION and DEPLOYING_CLUSTER

File: core/src/main/scala/org/apache/spark/internal/config/Deploy.scala [hunk 2]
```diff
@@ -20,58 +20,69 @@ package org.apache.spark.internal.config
 private[spark] object Deploy {
   val RECOVERY_MODE = ConfigBuilder("spark.deploy.recoveryMode")
     .version("0.8.1")
+    .scope(EffectTiming.DEPLOYING_CLUSTER)
```
Comment (dohongdayi) [inline core/src/main/scala/org/apache/spark/internal/config/Deploy.scala:23]: How about `.effectTiming(DEPLOYING_CLUSTER)` ?

**Pull request closed without merging**